### ISSUE TITLE
Support for enabling/disabling dual-stack

### ISSUE THREAD

**Issue by ben221199** [webfolderio/libuv-java#6]
When I listen on `0.0.0.0`, I actually listen for all IPv4 addresses. However, when I listen on `::`, it depends if `bindipv6only` is enabled. If it is, it only listens for all IPv6 addresses. If it is disabled, it listens for both IPv4 and IPv6.

In my case I want to only listen on `0.0.0.0` and `::` seperately. However, `bindipv6only` is enabled somehow and I cannot find a way to disable it in combination with this library.

Is it possible to make a function in the library to enable/disable the `bindipv6only` option when using this library in a project?

**Comment by webfolderio** [comment]
Libuv has `UV_UDP_IPV6ONLY` flag to disable/enable feature which is not supported by libuv-java?

Do you need such feature? If yes, I could add this option to libuv-java.

http://docs.libuv.org/en/v1.x/guide/networking.html#ipv6-stack-only

**Comment by ben221199** [comment]
Well, the problem occurs with TCP.

**Comment by ben221199** [comment]
I see your commit. Instead of using a boolean as parameter for only `ipv6only`, isn't it easier to use a `flags` parameter for using more flags than only `ipv6only`?

**Comment by webfolderio** [comment]
@ben221199 Could you please download the necessary artificats from the below link?

I tried to fix the problem with this commit: https://github.com/webfolderio/libuv-java/commit/25769d88e74aae808f2a0343202807d77228017a

Do not forgot to set `ipv6` flag to [bind](https://github.com/webfolderio/libuv-java/blob/master/src/main/java/com/oracle/libuv/TCPHandle.java#L35) method.

```java
handle.bind("localhost", 8080, true); // IPv6 stack only
// http://docs.libuv.org/en/v1.x/guide/networking.html#ipv6-stack-only
```

https://github.com/webfolderio/libuv-java/actions/runs/1066637108

**Comment by ben221199** [comment]
I still get the same error. I have the following code in my build.gradle:
```
implementation 'com.github.webfolderio:libuv-java:f9cd04d1ac'
```

I think the natives should be compiled again.

**Comment by webfolderio** [comment]
> I see your commit. Instead of using a boolean as parameter for only ipv6only, isn't it easier to use a flags parameter for using more flags than only ipv6only?

Yes, it's better to add an extra flags argument to `bind` method. I will do it.

> I think the natives should be compiled again.

What is your OS? Mac, Windows, Ubuntu/CentOS...

**Comment by ben221199** [comment]
I'm developing on Windows, but my production server is on Linux (one Ubuntu, other CentOS).

**Comment by ben221199** [comment]
Well okay, you used enum, but you could have used an integer too in Java. At least, I now have the possibility to use flags.

**Comment by webfolderio** [comment]
> Well okay, you used enum, but you could have used an integer too in Java.

I try my best to sync enum values with libuv tcp/udp flags. I might add an extra bind method to support `int` type also.

**Comment by ben221199** [comment]
Ah, ok.

**Comment by ben221199** [comment]
When will there be a new release to easily import in my Maven/Gradle?

**Comment by webfolderio** [comment]
> When will there be a new release to easily import in my Maven/Gradle?

Probably tomorrow.

**Comment by ben221199** [comment]
Woops:
![image](https://user-images.githubusercontent.com/12856904/127217143-705a1667-73f4-4758-9270-5df0da034b7d.png)

**Comment by webfolderio** [comment]
@ben221199 I published the latest version 1.1.0 maven central. Could you please check if it works?

**Comment by ben221199** [comment]
On Windows:
![image](https://user-images.githubusercontent.com/12856904/127370517-e9bc66a6-1003-46a3-8821-a3a24173efe7.png)

On Linux:
![image](https://user-images.githubusercontent.com/12856904/127370936-0d57df94-1ece-4e5a-b5ab-838cf41b327d.png)

**Comment by ben221199** [comment]
You should fix this.

BTW, if I'm testing my server with Netty, I get much better performance. 🤔

**Comment by webfolderio** [comment]
@ben221199 Could you please share a simple test case to reproduce the error/bug?

**Comment by ben221199** [comment]
Just this:
```java
LoopHandle LOOP = new LoopHandle();
HandleFactory FACTORY = new DefaultHandleFactory(LOOP);

TCPHandle server = FACTORY.newTCPHandle();
server.setSimultaneousAccepts(true);
server.setConnectionCallback(new MyTCPServerSocketCallback(server));
server.bind(IPv4,53,false,EnumSet.of(TcpFlags.UV_TCP_IPV6ONLY));
server.listen(0);
```

**Comment by ben221199** [comment]
I have 4 functions that start in the following sequence:
 - TCPv4
 - TCPv6
 - UDPv4
 - UDPv6

Because TCPv4 is the first one, there is no other port already in use, so that could not be the issue.

**Comment by webfolderio** [comment]
`Exception in thread "main" NativeException [errno=-4071, errnoString=EINVAL, errnoMessage=invalid argument, syscall=uv_tcp_bind, path=null]`

libuv-java throws above exception if you try to use `TcpFlags.UV_TCP_IPV6ONLY` with `ipv6=true`

The below code snippet **throws exception**:
```java
boolean ipv6 = false;
server.bind("localhost", 53, ipv6, EnumSet.of(TcpFlags.UV_TCP_IPV6ONLY));
```

The below code snippet works correctly:
```java
boolean ipv6 = true;
server.bind("::1", 53, ipv6, EnumSet.of(TcpFlags.UV_TCP_IPV6ONLY));
```

> I have 4 functions that start in the following sequence: TCPv4, TCPv6, UDPv4, UDPv6

libuv supports [dual-stack](http://docs.libuv.org/en/v1.x/tcp.html#c.uv_tcp_bind) by default. You do not need to invoke bind method more than once.

if you want to listen localhost host for both **ipv4 and ipv6**, use the below code snippet:

```java
boolean ipv6 = true;
server.bind("localhost", 53, ipv6);
```

if you want to listen localhost host for only **ipv4**, use the below code snippet:

```java
boolean ipv6 = false;
server.bind("localhost", 53, ipv6);
```

if you want to listen localhost host for only **ipv6**, use the below code snippet:

```java
boolean ipv6 = true;
server.bind("::1", 53, ipv6, EnumSet.of(TcpFlags.UV_TCP_IPV6ONLY));
```

[http://docs.libuv.org/en/v1.x/guide/networking.html#ipv6-stack-only](http://docs.libuv.org/en/v1.x/guide/networking.html#ipv6-stack-only)
> **IPv6 sockets can be used for both IPv4 and IPv6 communication.** If you want to restrict the socket to IPv6 only, pass the UV_UDP_IPV6ONLY flag to uv_udp_bind 2.
